compiler/damlc/tests/src/VisualTest.hs

```diff
@@ -0,0 +1,111 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+module VisualTest
+   ( main
+   ) where
+import DA.Cli.Visual
+import Data.Either
+import qualified Test.Tasty.Extended as Tasty
+import qualified Test.Tasty.HUnit    as Tasty
+import qualified DA.Pretty as DAP
+import qualified Development.IDE.Types.Location as D
+import qualified DA.Daml.LF.Ast as LF
+import qualified Development.IDE.Core.API as API
+import qualified Development.IDE.Core.Rules.Daml as API
+import qualified Data.NameMap as NM
+import qualified Control.Monad.Reader   as Reader
+import Development.IDE.Core.API.Testing
+import System.Environment.Blank (setEnv)
+import Control.Monad.Except
+
+data ExpectedChoices = ExpectedChoices
+    { _cName :: String
+    , _consuming :: Bool
+    } deriving (Eq, Show )
+data TemplateProp = TemplateProp
+    { _choices :: [ExpectedChoices]
+    , _action :: Int
+    } deriving (Eq, Show)
+
+main :: IO ()
+main = do
+    setEnv "TASTY_NUM_THREADS" "1" True
+    Tasty.deterministicMain visualTests
+
+visualTests :: Tasty.TestTree
+visualTests =
+    Tasty.testGroup "Visual tests using Shake API" [ visualDamlTests ]
+
+testCase :: Tasty.TestName -> ShakeTest () -> Tasty.TestTree
+testCase testName test =
+    Tasty.testCase testName $ do
+        res <- runShakeTest Nothing test
+        Tasty.assertBool ("Shake test resulted in an error: " ++ show res) $ isRight res
+
+templateChoicesToProps :: TemplateChoices -> TemplateProp
+templateChoicesToProps tca  = TemplateProp choicesInTpl actl
+    where choicesInTpl = map (\ca -> ExpectedChoices ( DAP.renderPretty $ choiceName ca) (choiceConsuming ca)) (choiceAndActions tca)
+          actl = sum $ map (length . actions ) (choiceAndActions tca)
+
+graphTest :: LF.World -> LF.Package -> [TemplateProp] -> Either (String, String) ()
+graphTest wrld lfPkg tplProps =
+    if tplProps == map templateChoicesToProps tplPropsActual
```

Comment:
Does the order matter here or do we only want to test that the lists have the same elements?

Reply:
No order should not matter, I will change it to Set